xcdat/temporal.py

```diff
@@ -936,8 +954,8 @@ def _group_average(self, data_var: xr.DataArray) -> xr.DataArray:
         # attributes are removed. Xarray's `keep_attrs=True` option only keeps
         # attributes for data variables and not their coordinates, so the
         # coordinate attributes have to be restored manually.
-        dv[self._dim].attrs = self._labeled_time.attrs
-        dv[self._dim].encoding = self._labeled_time.encoding
+        dv[self._dim].attrs = lt_attrs
+        dv[self._dim].encoding = lt_encoding
```

Comment:
These lines used to be 

```
dv[self._dim].attrs = self._labeled_time.attrs
dv[self._dim].encoding = self._labeled_time.encoding
```

Somehow this information was getting dropped in the new weighted average call (`dv = self._group_data(dv).sum() / self._group_data(weights).sum()`). 

I'm not sure if this can be improved...I store `lt_attrs` and `lt_encoding` in two places: under `if self._weighted:` and `else` because these values are only populated after `._group_data()` (in the `else` statement these values are populated after calling `_get_weights()` because `_get_weights()` calls `_group_data()`.

Reply:
Refactored this in my recent review